### PR TITLE
manual for 2.16, fixes #2265

### LINKED ISSUE #2265: next manual, 2.15++
#### Summary:

Corrections and suggestions for the manual after the 2.15 release should be included as comments.  

The branch for this issue is

* `feature/2265-manual-2.16`

#### Current Version:
v2.15.0

### PR DESCRIPTION
**Pull request by bob-carpenter**
#### Submisison Checklist

- [x] Declare copyright holder and open-source license: see below

#### Summary

Fix all manual issues in #2265 

#### Intended Effect

Enable release of 2.16.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/docs/stan-reference/programming.tex [hunk 1]
```diff
@@ -514,10 +514,35 @@ \section{Vectors with Varying Bounds}
...
+  ...
+transformed parameters {
+  vector[N] alpha = L + alpha_raw;
+  ...
+\end{stancode}
+%
+The Jacobian for adding a constant is one, so its log drops out of the
+log density.
+
+Even if the lower bound is a paramter rather than data, there is no
```
Comment (syclik) [inline src/docs/stan-reference/programming.tex]: typo: "paramter" -> "parameter"

File: src/docs/stan-reference/programming.tex [hunk 2]
```diff
@@ -546,6 +571,33 @@ \section{Vectors with Varying Bounds}
 $\alpha_{\mathrm{raw}}$ then a revised Jacobian needs to be calculated
 taking into account the dependencies.
 
+\subsection{Varying Upper and Lower Bounds}
+
+Suppowe there are lower and upper bounds that vary by parameter.
```
Comment (syclik) [inline src/docs/stan-reference/programming.tex]: typo: "Suppowe" -> "Suppose"

**Comment by syclik** [comment]
@bob-carpenter, I just pushed a commit fixing the spelling errors I spotted in this pull request and a few others scattered throughout the manual. Feel free to merge when ready.

**Comment by bob-carpenter** [comment]
Thanks!  I'll merge when the tests finish.  Is there perhaps a way to reduce the number of tests when only the manual changes?

**Comment by syclik** [comment]
Yes, I'm sure we can figure that out.

**Pull request merged**